inventory/linux.cf

```diff
@@ -41,23 +41,18 @@ bundle common inventory_linux
         if => strcmp("$(proc_routes[$(routeidx)][1])", "00000000");
 
     linux::
-      "nfs_servers" -> { "CFE-3259" }
-        comment => "NFS servers (to list hosts impacted by NFS outages)",
-        slist => maplist( regex_replace( $(this) , ":.*", "", "g"),
-                          # NFS server is before the colon (:), that's all we want
-                          # e.g., nfs.example.com:/vol/homedir/user1 /home/user1 ...
-                          #       ^^^^^^^^^^^^^^^
-                          grep( ".* nfs .*",
-                                readstringlist("/proc/$(this.promiser_pid)/mounts", "", "\n", inf, inf)
-                              )
-                        ),
-        if => and( fileexists( "/proc/$(this.promiser_pid)/mounts" ),
-                   not( isvariable( "$(this.promiser)" ) ) );
-
-
-        "nfs_server[$(nfs_servers)]"
-          string => "$(nfs_servers)",
-          meta => { "inventory", "attribute_name=NFS Server" };
+
+      "mounts" string => "/proc/$(this.promiser_pid)/mounts";
+
+      "nfs_mounts"
+        slist => grep( ".* nfs .*",
+                       readstringlist("$(mounts)", "", "\n", inf, inf) ),
+        if => not( isvariable( "$(this.promiser)" ) );
```

Comment:
I don't think we should solve this problem in this way. If defining a variable usually happens multiple times, we should fix that.

Reply:
You also need to keep the guard on the file existing right?

Reply:
> I don't think we should solve this problem in this way. If defining a variable usually happens multiple times, we should fix that.

Maybe not. But it's way more performant. This fix has the benefit of working with existing binaries.

> You also need to keep the guard on the file existing right?

Yeah, probably should on the one that readstringlists